src/main/resources/templates/fragments/inputs/date.html

```diff
@@ -3,14 +3,30 @@
     th:with="
       hasLabel=${!#strings.isEmpty(label)},
       hasAriaLabel=${!#strings.isEmpty(ariaLabel)},
+      isRequiredInput=${requiredInputs.getOrDefault(inputName + 'Month', false) || 
+        requiredInputs.getOrDefault(inputName + 'Day', false) || 
+        requiredInputs.getOrDefault(inputName + 'Year', false) ||
+        required != null && required},
```

Comment:
I would add parenthesis around the `required` part, like so `(required != null && required)` to make sure it's clear how to interpret this.  Just makes it a little easier to read and very clear that those work together.

Reply:
Done